course/models.py

```diff
@@ -790,6 +801,7 @@ def enroll_student(self, user):
                     return False
```

Comment:
If the student already has an active enrollment, the sis flag is not enabled here during enroll_from_sis. Maybe we should check the sis flag here and enable or disable it for existing enrollments too. Why not flag all sis participants in Aplus too?

Reply:
The only current reason for from_sis flag is to support the following scenario: if student has been enrolled based on SIS, and if she later unenroll's in SIS, then A+ enrollment should also be removed, to keep these two in sync. If a student had enrolled directly in A+ (e.g. for being a MOOC student), then she should not be unenrolled based on changes in SIS data.

Ideally local students should not be able to enroll directly when SIS is in use. A follow-up PR adds option to disallow direct enrolling by local students, but when a course has both MOOC and local students, then we need to know the enrollment method to support automatic unenrolling. So in summary: I think we should leave this as it is.

Reply:
I meant that if a student has enrolled locally in A+, but his/her enrollment exists in Sisu too, then the enrollment could be flagged to the Sisu mode in A+. Then, the student would be able to uneroll in Sisu (which is propagated to A+) and if we include only Sisu students in the Sisu grades CSV export, then we need to know all students that exist in Sisu.

If local students are not able to enroll locally in A+ (only in Sisu), then the above scenario would never occur. Unless teachers enroll late students manually in A+ since Sisu has been already closed...

Reply:
ok, accepted that it is better to do as you suggest